app/data/generators/rejection.js

```diff
@@ -1,98 +1,70 @@
-    'other-feedback': 'No'
-  }
+function buildReasons(params) {
 
-  const behaviour = {
-    actions: 'Yes',
-    'actions-reasons': [
-      'Did not reply to messages',
-      'Other'
-    ],
-    'actions-reasons-other': faker.lorem.paragraph(),
-    'actions-reasons-other-improve': faker.lorem.paragraph()
+  let o = {
+    categories: []
   }
 
-  const quality = {
-    'application-quality': 'Yes',
-    'application-quality-reasons': [
-      'Personal statement',
-      'Subject knowledge'
-    ],
-    'application-quality-reasons-personal-statement': faker.lorem.paragraph(),
-    'application-quality-reasons-subject-knowledge': faker.lorem.paragraph()
-  }
+  if(params.qualifications) {
+    o.categories.push('Qualifications')
 
-  const qualifications = {
-    'missing-qualifications': 'Yes',
-    'missing-qualifications-reasons': [
-      'No maths GCSE grade 4 (C) or above, or accepted equivalent',
+    o['qualifications-reasons'] = [
       'No English GCSE grade 4 (C) or above, or accepted equivalent',
+      'Degree does not meet course requirements',
       'Other'
-    ],
-    'missing-qualifications-reasons-other': faker.lorem.paragraph(1)
-  }
+    ]
 
-  const interviewPerformance = {
-    'interview-performance': 'Yes',
-    'interview-performance-advice': faker.lorem.paragraph()
-  }
+    if(o['qualifications-reasons'].find(item => 'Degree does not meet course requirements')) {
+      o['qualifications-reasons-degree-does-not-meet-course-requirements'] = faker.lorem.paragraph(1)
+    }
 
-  const course = {
-    'course-full': 'Yes'
-  }
 
-  const offeredOther = {
-    'other-offer': 'Yes',
-    'other-offer-details': faker.lorem.paragraph()
+    if(o['qualifications-reasons'].find(item => 'Other')) {
+      o['qualifications-reasons-other'] = faker.lorem.paragraph(1)
+    }
+
   }
 
-  const honesty = {
-    honesty: 'Yes',
-    'honesty-reasons': [
-      'Inaccurate or false information in the application',
-      'Evidence of plagiarism in the application',
+  if(params.personalStatement) {
+    o.categories.push('Personal statement')
+
+    o['personal-statement-reasons'] = [
+      'Quality of writing',
       'Other'
-    ],
-    'honesty-reasons-false-information': faker.lorem.paragraph(1),
-    'honesty-reasons-plagiarism': faker.lorem.paragraph(1),
-    'honesty-reasons-other': faker.lorem.paragraph(2)
-  }
+    ]
 
-  const safeguarding = {
-    safeguarding: 'Yes',
-    'safeguarding-reasons': [
-      'The vetting process found information which makes the candidate unsuitable to work with children'
-    ],
-    'safeguarding-reasons-vetting-information': faker.lorem.paragraph()
-  }
+    if(o['personal-statement-reasons'].find(item => 'Quality of writing')) {
+      o['personal-statement-reasons-quality-of-writing'] = faker.lorem.paragraph(1)
+    }
+
+    if(o['personal-statement-reasons'].find(item => 'Other')) {
+      o['personal-statement-reasons-other'] = faker.lorem.paragraph(1)
+    }
```

Comment:
Same again. Why do you need the if-statement if the item is always in the reasons array? Will you be randomising the reasons?

Reply:
As above

Reply:
Yes, that's fine